### PR TITLE
Ntseng/issue6

### LINKED ISSUE #6: Investigate giving club leaders event start permission
Was probably user error, but should probably add a "Start Event" button to the reminder message.

### PR DESCRIPTION
**Pull request by ntseng**
Summary
-------
- fixed a crash in changeclubmeeting
- changed club repeat period enums to the full word ("days" or "weeks")
- add Start Event button to reminder message
- cleaned up reminder and event scheduling logic
- added permissions check to `/club-config`

Local Tests Performed
---------------------
- open changeclubmeeting while nextMeeting is null
- used changeclubmeeting modal (will need migration for repeat period enums)
- started bot with next meetings set
- changed next meeting value via modal
- used /club-send-reminder

Issue
-----
Closes #6

### CODE REVIEW

File: source/modalSubmissions/changeclubmeeting.js [hunk 1]
```diff
@@ -16,17 +18,30 @@ module.exports = new ModalSubmission(id,
...
-			if (nextMeetingInput) {
-				club.timeslot.setNextMeeting(nextMeetingInput);
-				clearClubReminder(club.id);
-				cancelClubEvent(club.voiceChannelId, club.timeslot.eventId, interaction.guild.scheduledEvents);
-				setClubReminder(club, interaction.guild.channels);
-				createClubEvent(club, interaction.guild);
-				if (club.timeslot.periodCount && club.isRecruiting()) {
-					scheduleClubEvent(club, interaction.guild);
-				}
+			if (!nextMeetingInput) {
```
Comment (SillySalamndr) [inline source/modalSubmissions/changeclubmeeting.js:21]: The vast majority of the start of this function is a _lot_ of input validation and parsing, which could be cleaned up with some early-out-like behavior. Additionally, it seems like if some of the input fails validation, we still call other helper functions to execute on the action before we display the errors to the user and leave, which seems wrong.

File: source/helpers.js [hunk 2]
```diff
@@ -767,7 +753,6 @@ exports.setClubReminder = async function (club, channelManager) {
 
 /** The number of ms until a day before the next meeting, but not more than the max allowable setTimeout duration
  * @param {number} timestamp The unix timestamp of the next meeting (in seconds)
- * @returns
```
Comment (SillySalamndr) [inline source/helpers.js:769]: Was removing this line in the JSDocs intentional?

File: source/buttons/startevent.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+						console.error(`Event ${eventId} could not be started because the event was not found.`);
+						interaction.reply({ content: "HorizonsBot could not find the event, please start the event manually or contact a moderator.", ephemeral: true });
+						break;
+					default:
+						console.error(error);
+				}
+				interaction.message.edit({ components: [] });
+			});
+		} else {
+			interaction.reply({ content: "Only the club's leader or a moderator can start the event.", ephemeral: true });
```
Comment (SillySalamndr) [inline source/buttons/startevent.js]: Moving this validation error to the beginning as an early out would help tame the pretty intense amount of indentation in this function.

File: source/helpers.js (hunk 2 above)
Comment (ntseng) [reply source/helpers.js:769]: Yeah, type annotations were able to detect that the function was returning a number without it.

File: source/modalSubmissions/changeclubmeeting.js (hunk 1 above)
Comment (ntseng) [reply source/modalSubmissions/changeclubmeeting.js:21]: The only validations we can use before submission are based on string length or existence, so our checks for ability to parse to integer and ranges have to be in-app.

In the case where the modal receives some appropriate data and some errors that can't be caught prior to submission, I think it's best to still process the appropriate data.

File: source/buttons/startevent.js (hunk 3 above)
Comment (ntseng) [reply source/buttons/startevent.js]: Added in https://github.com/Imaginary-Horizons-Productions/HorizonsBot/pull/44/commits/d87e3b6be573ee6843ceeed898d66572bcc0466f

**Pull request merged**